lib/Net/SSLeay.pod

```diff
@@ -4625,6 +4625,32 @@ When callback is undef, an existing callback will be disabled.
 
 Check openssl doc L<http://www.openssl.org/docs/ssl/SSL_CTX_set_info_callback.html|http://www.openssl.org/docs/ssl/SSL_CTX_set_info_callback.html>
 
+=item * set_msg_callback
+
+Sets the callback function, that can be used to obtain protocol messages information for $ssl during connection setup and use.
+When callback is undef, the callback setting currently valid for ctx is used.
+
+ Net::SSLeay::set_msg_callback($ssl, $cb, [$data]);
+ # $ssl - value corresponding to openssl's SSL structure
+ # $cb - sub { my ($ssl,$write_p,$version,$content_type,$buffer,$len,$data) = @_; ... }
```

Comment:
OpenSSL's callback uses the arguments in this order and with these names:
`(int write_p, int version, int content_type, const void *buf, size_t len, SSL *ssl, void *arg)`

Can you update the argument list to match what OpenSSL uses?

Reply:
I've put $ssl first, as that's the same for all the other callbacks, and *arg isn't needed/used. I'll just rename "buffer" to "buf"?

Reply:
Thanks for your patience with my suggestions! What I'd still like to see is that argument order matches what OpenSSL uses. I think the other callbacks have SSL first simply because that's what OpenSSL does too. What I'd like to do is to deviate from OpenSSL as little as possible.

If we update `$ssl`'s position that would only leave the callback `$data / *arg`, which includes not needing `SSL_*set_msg_callback_arg`, as the only differences to how OpenSSL has documented this.

Reply:
Np! Glad that there's feedback to improve things a bit.
I've documented this now, and changed $data into $arg to make this more clear.